### PR TITLE
Fix, workaround for maybe uninitialized variables

### PR DESCRIPTION
**Pull request by dexX7**
Warnings addressed:

mastercore_rpc.cpp:1515:10: warning: ‘valid’ may be used uninitialized
in this function [-Wmaybe-uninitialized]
mastercore_rpc.cpp:1845:9: warning: ‘MPTxTypeInt’ may be used
uninitialized in this function [-Wmaybe-uninitialized]

I'm on another PR that aims to collect constants etc., so please don't
mind the #define..

### CODE REVIEW

**Comment by dexX7** [comment]
Note: rebased. Now it does, what it should.

**Comment by m21** [comment]
@zathras-crypto @faizkhan00 not sure how they got introduced, thanks Dexx for noticing.

I am against adding new types to general-purpose global code (TYPE_UNDEFINED) to simply help with this RPC function. There is a logic problem in it somewhere and should be fixed locally.
Probably same for "valid" -- it can be simply initialized to "false", IF it makes logical sense please. :)

**Comment by dexX7** [comment]
Breaking populateRPCTransactionObject into parts and defining variables more locally would help.

But you were right, I found another path which builds without warning - PR updated.

File: src/mastercore_rpc.cpp [hunk 1]
```diff
@@ -1784,6 +1784,10 @@ static int populateRPCTransactionObject(uint256 txid, Object *txobj, string filt
                     } // end switch
                     divisible=isPropertyDivisible(propertyId);
                 } // end step 1 if
+                else
+                {
+                    return -3336; // "Not a Master Protocol transaction"
```
Comment (m21) [inline src/mastercore_rpc.cpp:1789]: Another magic number :)
Does it belong to MPRPCErrorCode perhaps?

**Pull request merged**